### DISCUSSION TITLE
Could you please help me to solve this equation.

### DISCUSSION THREAD

**Discussion by ztdepztdep** [ginkgo-project/ginkgo#1043] [Q&A] [answered]
[AA3.txt](https://github.com/ginkgo-project/ginkgo/files/8751769/AA3.txt)
[Ab3.txt](https://github.com/ginkgo-project/ginkgo/files/8751781/Ab3.txt)
This A and b were generated with the spectral element method, this is a simple heat diffusion equation and it has an exact solution of 25.0. 
I have tried many solver, but i cann't get a correct one. Could you please help me out.

**Comment by fritzgoebel** [comment, marked as answer]
I assume with "exact solution of 25.0" you mean a vector with all entries equal to 25.0?
If so, I suspect something in your problem generation might be going wrong since with the A and b you provided and x filled with entries of 25.0 I get `||A*x-b|| = 306.1862`.
If not, please let me know. For the matrix and right hand side you provided, plain Gmres and Bicgstab produce accurate solutions for me.

**Reply by ztdepztdep** [reply]
Thanks you very much .
I output the vector from the Trilinos package by API     EpetraExt::RowMatrixToMatrixMarketFile("./output/AA3.mtx",*VarBox_RCP->A_Diffusion,"A","A",TRUE);
EpetraExt::VectorToMatrixMarketFile("./output/Ab3.mtx",S2,"B","B",TRUE);
this problem can be solved exactly by the Trilinos package with gmres method. I am really confused by this situations. 
Does the input file has some specified format?
Regards

**Reply by fritzgoebel** [reply]
Yes, Gmres is able to solve the problem accurately. But for the matrix you provided here, the solution is not a vector with all entries equal to 25. You can easily check this e.g. by loading your problem into Matlab. Maybe in your code using Trilinos there is some boundary condition treatment happening that's not reflected in the files you uploaded here?

**Reply by ztdepztdep** [reply]
Great. I found the problems. the vector file output by the Trilinos is not in the order of matrix rows. Thank you very much for your help.
